tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +40,196 @@ class resource_type(ArmResourceManager.resource_type):
+                    op: lt
+                    retention-period-days: 14
+
+    """
+
+    schema = type_schema(
+        'short-term-backup-retention-policy',
+        required=['retention-period-days'],
+        rinherit=BackupRetentionPolicyFilter.schema,
+        **{
+            'retention-period-days': {'type': 'number'}
+        }
+    )
+
+    def __init__(self, data, manager=None):
+        retention_limit = data.get('retention-period-days')
+        super(ShortTermBackupRetentionPolicyFilter, self).__init__(
+            'backup_short_term_retention_policies', retention_limit, data, manager)
+
+    def get_retention_from_policy(self, retention_policy):
+        return retention_policy.retention_days
+
+
+@SqlDatabase.filter_registry.register('long-term-backup-retention-policy')
+class LongTermBackupRetentionPolicyFilter(BackupRetentionPolicyFilter):
+    """
+
+    Filter SQL Databases on the length of their long term backup retention policies.
+
+    There are 3 backup types for a sql database: weekly, monthly, and yearly. And, each
+    of these backups has a retention period that can specified in units of days, weeks,
+    months, or years.
+
+    :example: Find all SQL Databases with weekly backup retentions longer than 1 month.
+
+    .. code-block:: yaml
+
+            policies:
+              - name: long-term-backup-retention-policy
+                resource: azure.sqldatabase
+                filters:
+                  - type: long-term-backup-retention-policy
+                    backup-type: weekly
+                    op: gt
+                    retention-period: 1
+                    retention-period-units: months
+
+    """
+
+    @enum.unique
+    class BackupType(enum.Enum):
+        weekly = 'weekly'
+        monthly = 'monthly'
+        yearly = 'yearly'
+
+        def __str__(self):
+            return self.value
+
+    schema = type_schema(
+        'long-term-backup-retention-policy',
+        required=['backup-type', 'retention-period', 'retention-period-units'],
+        rinherit=BackupRetentionPolicyFilter.schema,
+        **{
+            'backup-type': {'enum': list([str(t) for t in BackupType])},
+            'retention-period': {'type': 'number'},
+            'retention-period-units': {
+                'enum': list([str(u) for u in RetentionPeriod.Units])
+            }
+        }
+    )
+
+    def __init__(self, data, manager=None):
+        retention_period = data.get('retention-period')
+        self.retention_period_units = RetentionPeriod.Units[
+            data.get('retention-period-units')]
+
+        super(LongTermBackupRetentionPolicyFilter, self).__init__(
+            'backup_long_term_retention_policies', retention_period, data, manager)
+        self.backup_type = self.data.get('backup-type')
+
+    def get_retention_from_policy(self, retention_policy):
+        if self.backup_type == LongTermBackupRetentionPolicyFilter.BackupType.weekly.value:
```

Comment:
nit: any chance we can make it in some standard way? i.e. if you extend enum, you don't have to update code anywhere else.
I like your enum approach, so it might be useful for other places.

Reply:
yeah, I can expand the enum a bit to reduce the if, elif, elif, ... chain going on here.